src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -577,6 +577,11 @@ private ConnectionCompleteParams GetConnectionCompleteParams(string connectionTy
                 DbConnection underlyingConnection = reliableConnection != null
                     ? reliableConnection.GetUnderlyingConnection()
                     : connection;
+                
+                var pid = (underlyingConnection as SqlConnection).ServerProcessId;
+                if (pid != 0) {

```

Comment:
nit: A comment here explaining that 0 means the connection is inactive would be helpful for those that aren't aware